scenario/state.py

```diff
@@ -140,6 +141,77 @@ def copy(self) -> "Self":
         return copy.deepcopy(self)
 
 
+@dataclasses.dataclass(frozen=True)
+class CloudCredential:
+    auth_type: str
+    """Authentication type."""
+
+    attributes: Dict[str, str] = dataclasses.field(default_factory=dict)
+    """A dictionary containing cloud credentials.
+
+    For example, for AWS, it contains `access-key` and `secret-key`;
+    for Azure, `application-id`, `application-password` and `subscription-id`
+    can be found here.
+    """
+
+    redacted: List[str] = dataclasses.field(default_factory=list)
+    """A list of redacted secrets."""
```

Comment:
this sounds mysterious to me. Can we be more explicit or provide an example? Are we talking juju secret IDs, or generic cloud API secret keys?

Reply:
This is CloudCredential, whose name suggests it's cloud API secrets. I think it's straightforward enough, and it's consistent with CloudCredential in the ops framework. I'm OK if we want to make it more verbose with examples, but I think it'd be nice to update ops framework first then scenario follows.

Reply:
For what it's worth, I was a bit hesitant about this in the ops docs too, and I probably should have mentioned that there.

I think Pietro's suggestion of an example (both here and in ops) would probably really help.

Reply:
let's do this, then we can close! :)